tools/project_manager/diff.php

```diff
@@ -138,11 +141,26 @@
     echo "| <a href='$format_url'>$format_label</a>\n";
 }
 
+if (user_can_mentor_in_any_round()) {
+    // show option to change diff style
+    if ($bb_diffs) {
+        $bb_diffs_label = _("Table diff style");
+        $_GET["bb_diffs"] = "off";
+    } else {
+        $bb_diffs_label = _("BB diff style");
+        $_GET["bb_diffs"] = "on";
+    }
+    $bb_diffs_url = "?" . attr_safe(http_build_query($_GET));
+    echo "| <a href='$bb_diffs_url'>$bb_diffs_label</a>\n";
+}
+
 echo "</p>\n";
 
 // ---------------------------------------------------------
 
-$diffEngine = new DifferenceEngineWrapper();
+$diffurl = $_SERVER["REQUEST_SCHEME"]."//".$_SERVER['HTTP_HOST'].$_SERVER["PHP_SELF"];
```

Comment:
Since the HTML will be rendered on this page, we should be able to use the relative URL (just the line below) and not need to reconstruct the full URL. Using relative URLs when we can is preferred. When we can't we use `$code_url` global instead (eg: `$code_url/tools/project_manager/diff.php`).

Reply:
It does need the full URL in this case, since it's not rendering on this page. It's passing the diffurl down into the DiffEngineWrapper and thence to the DiffFormatter. From there, it will be output into the BB-style diffs as `[url=$diffurl]001.png[/url]`, i.e. the URL is included in the PM to the mentee, so the full URL is needed. I've changed it back to use `$code_url` (as I had in my noncvs experiment). It's good to know about the relative URL thing though, thanks.

Reply:
Ah, but the *entry* page is `diff.php` and that's what the link will be submitting to -- the web browser doesn't know about anything else being included. I really think you can get away with a relative URL.

Reply:
Sorry, I didn't read your message clearly. Yup, I see now that we need to include the absolute URL and what you have here is good!